libs/api/domains/license-service/src/lib/licenseService.service.ts

```diff
@@ -567,7 +573,11 @@ export class LicenseServiceService {
       }),
     ])
 
-    return token
+    return {
+      token,
+      // Make sure to convert the expiration time to milliseconds
+      exp: new Date(exp * 1000),
```

Comment:
Is the expiration value in nanoseconds? 🤔

Reply:
Short answer, no the exp is in seconds because of the jwt lib.

Longer answer 😅 
Unfortunately the `jsonwebtoken` package does not support setting `exp` in milliseconds. I tried it without luck.
Here is a sample from their docs

```
// Signing a token with 1 hour of expiration:
jwt.sign({
  exp: Math.floor(Date.now() / 1000) + (60 * 60),
  data: 'foobar'
}, 'secret');
```

and this is how I am doing it to make sure the expiration date is exactly the same, e.g. what is used in the token and returned from the field resolver

```
  async createToken(data: LicenseTokenData): Promise<{
    token: string
    exp: number
  }> {
    const exp = Math.floor(Date.now() / 1000) + BARCODE_EXPIRE_TIME_IN_SEC

    return new Promise((resolve, reject) =>
      sign(
        {
          data,
          exp,
        },
        this.config.barcodeSecretKey,
        {},
        (err, encoded) => {
          if (err || !encoded) {
            return reject(err)
          }

          return resolve({
            token: encoded,
            exp,
          })
        },
      ),
    )
  }
```

However when seeing this I should rather multiply by 1000 within the func

Reply:
Sure. I just turned this around in my head. 👍